app/scripts/modules/core/src/presentation/tables/Table.tsx

```diff
@@ -0,0 +1,67 @@
+import * as React from 'react';
+
+import { useIsMobile } from 'core/presentation';
+
+import { ITableRowLayoutProps } from './TableRow';
+import { ITableCellLayoutProps } from './TableCell';
+import { TableContext } from './TableContext';
+
+const { useState, useMemo } = React;
+
+export interface ITableColumn {
+  name: string;
+  sortDirection?: 'ascending' | 'descending';
+  onSort?: () => any;
+}
+
+export type ITableColumns = Readonly<Array<string | ITableColumn>>;
+
+export type ITableLayoutProps = Omit<ITableProps, 'layout' | 'columns'> & {
+  columns: ITableColumn[];
+  isMobile: boolean;
+  expandAll: () => any;
+};
+
+export interface ITableLayout {
+  getTableLayout: (props: ITableLayoutProps) => JSX.Element;
+  getTableRowLayout: (props: ITableRowLayoutProps) => JSX.Element;
+  getTableCellLayout: (props: ITableCellLayoutProps) => JSX.Element;
+}
+
+export interface ITableProps {
+  layout: ITableLayout;
+  columns: ITableColumns;
+  expandable?: boolean;
+  children?: React.ReactNode;
+}
+
+export const Table = (props: ITableProps) => {
+  const isMobile = useIsMobile();
```

Comment:
This is ok for now, but I am still of the mindset that `isMobile` is a global state of the page. The table, as a subcomponent of the page, should be viewing this boolean, not setting it.

Reply:
Would your preference be that individual layout implementations ask for the value from `useIsMobile` separately? e.g. each `getXYZLayout` implementation would call `useIsMobile`?

After discussing it I'm not actually against that approach, but it does mean I'd probably have to do some performance optimization on `useIsMobile` sooner than anticipated.